File: leetcode/easy/BestTimeToBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        lowest = prices[0]
        max_profit = 0
        for i in range(1, len(prices)):
            price = prices[i]
            profit = price - lowest
            if max_profit < profit:
                max_profit = profit
            if lowest > price:
                lowest = price
        return max_profit

File: leetcode/easy/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) {
            return 0;
        }
        int lowest = prices[0];
        int maxProfit = 0;
        for (int i = 1; i < prices.length; i++) {
            int price = prices[i];
            int profit = price - lowest;
            if (maxProfit < profit) {
                maxProfit = profit;
            }
            if (lowest > price) {
                lowest = price;
            }
        }
        return maxProfit;
    }
}
